plugins/extractors/metabase/extractor.go

```diff
@@ -0,0 +1,213 @@
+package metabase
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	_ "github.com/lib/pq"
+	"github.com/odpf/meteor/core"
+	"github.com/odpf/meteor/core/extractor"
+	"github.com/odpf/meteor/plugins"
+	meteorMeta "github.com/odpf/meteor/proto/odpf/meta"
+	"github.com/odpf/meteor/utils"
+)
+
+var (
+	client = &http.Client{
+		Timeout: 2 * time.Second,
+	}
+	session_id = ""
```

Comment:
what is this value for? btw in the real application, this extractor will be instantiated multiple times, will this package variable be an issue?

Reply:
Before making the first request to a particular api, in Metabase we need to obtain a session id with our credentials. So I used it as a global variable to avoid passing it with every function.

I am not sure if it should be a problem

Reply:
there is a possibility that this extractor will be instantiated multiple times and will share this `session_id` variables.
You could utilise extractor internal state (private fields) to avoid passing it to every function.
e.g.
```
type Extractor {
  sessionID string
}
```